homeassistant/components/hyperion/light.py

```diff
@@ -157,158 +153,243 @@ def effect(self):
+                # Owner is the effect name.
+                # See: https://docs.hyperion-project.org/en/json/ServerInfo.html#priorities
+                self._set_internal_state(
+                    rgb_color=DEFAULT_COLOR, effect=visible_priority[const.KEY_OWNER]
+                )
+            elif componentid == const.KEY_COMPONENTID_COLOR:
+                self._set_internal_state(
+                    rgb_color=visible_priority[const.KEY_VALUE][const.KEY_RGB],
+                    effect=KEY_EFFECT_SOLID,
+                )
+            self._update_ha_state()
 
-    def json_request(self, request, wait_for_response=False):
-        """Communicate with the JSON server."""
-        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-        sock.settimeout(5)
-
-        try:
-            sock.connect((self._host, self._port))
-        except OSError:
-            sock.close()
-            return False
-
-        sock.send(bytearray(f"{json.dumps(request)}\n", "utf-8"))
-        try:
-            buf = sock.recv(4096)
-        except socket.timeout:
-            # Something is wrong, assume it's offline
-            sock.close()
-            return False
-
-        # Read until a newline or timeout
-        buffering = True
-        while buffering:
-            if "\n" in str(buf, "utf-8"):
-                response = str(buf, "utf-8").split("\n")[0]
-                buffering = False
-            else:
-                try:
-                    more = sock.recv(4096)
-                except socket.timeout:
-                    more = None
-                if not more:
-                    buffering = False
-                    response = str(buf, "utf-8")
-                else:
-                    buf += more
-
-        sock.close()
-        return json.loads(response)
+    def _update_effect_list(self, _=None):
+        """Update Hyperion effects."""
+        if not self._client.effects:
+            return
+        effect_list = []
+        for effect in self._client.effects or []:
+            if const.KEY_NAME in effect:
+                effect_list.append(effect[const.KEY_NAME])
+        if effect_list:
+            self._effect_list = effect_list
+            self._update_ha_state()
+
+    def _update_full_state(self):
+        """Update full Hyperion state."""
+        self._update_adjustment()
+        self._update_priorities()
+        self._update_effect_list()
+
+        _LOGGER.debug(
+            "Hyperion full state update: On=%s,Brightness=%i,Effect=%s "
+            "(%i effects total),Color=%s",
+            self.is_on,
+            self._brightness,
+            self._effect,
+            len(self._effect_list),
+            self._rgb_color,
+        )
+
+    def _update_connection(self, json):
+        """Update client connection state."""
+        if json.get(const.KEY_CONNECTED):
+            self._update_full_state()
+        self._update_ha_state()
+
+    async def async_setup(self):
```

Comment:
Probably inline this method in `async_added_to_hass`. But it would be good to not add extra state updates. The entity will update state by default after `async_added_to_hass` is done.